### PR TITLE
OKTA-571100 - Devices SDK: iOS JWT Bearer

### PR DESCRIPTION
**Pull request by IldarAbdullin-okta**
### Problem Analysis (Technical)
- JWT-Bearer grant type network API implementation
- retrieveMaintenanceToken API implementation

### CODE REVIEW

**Review by FeiChen-okta — APPROVED** [review]

File: Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift [hunk 1]
```diff
@@ -272,6 +276,63 @@ class AuthenticatorEnrollment: AuthenticatorEnrollmentProtocol {
...
+                                               methodEnrollmentId: methodEnrollmentId)
+        let assertion: String
+        do {
+            assertion = try jwtGenerator.generate(with: "JWT", kid: signingKeyTag, for: bearerJWT, with: signingKey, using: .ES256)
+        } catch {
+            completion(.failure(DeviceAuthenticatorError.oktaError(from: error)))
+            return
+        }
+
+        let oidcClientId = policy.metadata.settings?.oauthClientId ?? ""
```
Comment (moisesolmedo-okta) [inline Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift:311]: should we guard and return error if oidcClientId is empty?

File: Sources/DeviceAuthenticator/Networking/ServerAPIProtocol.swift [hunk 2]
```diff
@@ -149,6 +155,50 @@ extension ServerAPIProtocol {
...
+                                  oidcClientId: String,
+                                  scopes: [String],
+                                  assertion: String,
+                                  completion: @escaping (Result<HTTPURLResult, DeviceAuthenticatorError>) -> Void) {
+        let completeURL = orgURL.appendingPathComponent("/oauth2/v1/token")
+        let contentTypeHeaderValue = "application/x-www-form-urlencoded"
+        let acceptHeaderValue = "application/json"
+        let grantType = "urn:ietf:params:oauth:grant-type:jwt-bearer"
+        let scopesString = scopes.joined(separator: " ")
+        var completePayload = "grant_type=" + grantType + "&" + "client_id=" + oidcClientId + "&" + "scope=" + scopesString + "&" + "assertion=" + assertion
```
Comment (moisesolmedo-okta) [inline Sources/DeviceAuthenticator/Networking/ServerAPIProtocol.swift:175]: should you use URLQueryItems instead?

Comment (IldarAbdullin-okta) [reply Sources/DeviceAuthenticator/Networking/ServerAPIProtocol.swift:175]: It is not query parameters. It is http body and I think it is pretty readable

File: Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift (hunk 1 above)
Comment (IldarAbdullin-okta) [reply Sources/DeviceAuthenticator/Enrollment/AuthenticatorEnrollment.swift:311]: will add

**Review by moisesolmedo-okta — APPROVED** [review]

**Pull request merged**